Core/GDCore/IDE/WholeProjectRefactorer.cpp

```diff
@@ -1058,21 +1037,15 @@ bool WholeProjectRefactorer::FixInvalidRequiredBehaviorProperties(
     auto& object = problem.GetSourceObject();
     auto suggestedBehaviorNames =
         GetBehaviorsWithType(object, problem.GetExpectedBehaviorTypeName());
-    auto& behaviorContent = problem.GetSourceBehaviorContent();
-    auto& behaviorMetadata = MetadataProvider::GetBehaviorMetadata(
-        project.GetCurrentPlatform(), behaviorContent.GetTypeName());
-    if (MetadataProvider::IsBadBehaviorMetadata(behaviorMetadata)) {
-      continue;
-    }
+    auto& behavior = problem.GetSourceBehaviorContent();
 
-    auto& behavior = behaviorMetadata.Get();
     if (suggestedBehaviorNames.empty()) {
       // No matching behavior on the object.
       // Add required behaviors on the object.
 
       auto& expectedBehaviorMetadata = MetadataProvider::GetBehaviorMetadata(
           project.GetCurrentPlatform(), problem.GetExpectedBehaviorTypeName());
-      if (MetadataProvider::IsBadBehaviorMetadata(behaviorMetadata)) {
+      if (MetadataProvider::IsBadBehaviorMetadata(expectedBehaviorMetadata)) {
```

Comment:
Yes, I guess that if:
- a user add a required behavior "B" property to a behavior "A"
- remove the behavior "B"
- add the behavior "A" to an object
It could have crashed the editor.

It's an interesting case to add I'll add a todo for this.